plugins/module_utils/foreman_helper.py

```diff
@@ -816,7 +827,11 @@ def _lookup_entity(self, identifier, entity_spec, params=None):
         try:
             if 'scope' in entity_spec:
                 for scope in entity_spec['scope']:
-                    params.update(self.scope_for(scope))
+                    if resource_type in ['content_views', 'repositories'] and scope == 'lifecycle_environment':
+                        scope_key = 'environment'
+                    else:
+                        scope_key = None
+                    params.update(self.scope_for(scope, scope_key))
```

Comment:
I like the `optional_scope` idea, but i would traverse it as a second loop here instead of fiddling with the `foreman_spec` depending on user input.